src/pages/view/FilterBar/useSaveFilterDialog.js

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useState } from 'react'
+
+export const useSaveFitlerDialog = ({
+    activeFilter,
+    doSaveFilter,
+    isLoading,
+}) => {
+    const [dialogIsOpen, setDialogIsOpen] = useState(false)
+    const [filterDialogData, setFilterDialogData] = useState(activeFilter)
+    const [showScope, setShowScope] = useState(true)
+
+    const openDialogForNewFilter = useCallback(() => {
+        setFilterDialogData({ id: 'new' })
```

Comment:
Can we create a constant for the 'new' string? So we can reuse it in both places. 

```ts
const NEW_FILTER_ID = 'new'; // feel free to use another name

...
setFilterDialogData({ id: NEW_FILTER_ID  })

...

const onConfirm = useCallback(
        ({ name, visibility, id }) => {
            doSaveFilter({
                ...activeFilter,
                name,
                visibility,
                id,
                ...(id === NEW_FILTER_ID  && { userName: null, userId: null }),
            }).then((resp) => resp && onClose())
        },
        [activeFilter, onClose, doSaveFilter]
    )
```

Just to be safe (in case we need to use the value elsewhere), we should reuse the variable. This will make refactoring easier if changes are needed in the future.